plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadataFactory.java

```diff
@@ -57,6 +62,12 @@ public IcebergMetadata create(ConnectorIdentity identity)
                 commitTaskCodec,
                 catalogFactory.create(identity),
                 fileSystemFactory,
-                tableStatisticsWriter);
+                tableStatisticsWriter,
+                metadataListingExecutor);
+    }
+
+    public void shutdown()
+    {
+        metadataListingExecutor.shutdown();
```

Comment:
metadataListingExecutor is a singleton, so should not be killed when metadata is torn down